### PR TITLE
Removed unused SSHKeys field and added comments on APIs

### PR DESCRIPTION
**Pull request by prashanth26**
**What this PR does / why we need it**:
- Removed unused SSHKeys field from API
- Added comments on API fields

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement user
Improved comments on API fields
```

/area documentation
/kind api-change
/priority critical

### CODE REVIEW

File: pkg/gcp/apis/provider_spec.go [hunk 1]
```diff
@@ -22,65 +22,257 @@ const (
...
+	// AutoDelete: Specifies whether the disk will be auto-deleted when the
+	// instance is deleted (but not when the disk is detached from the
+	// instance).
+	AutoDelete *bool `json:"autoDelete"`
+
+	// Boot: Indicates that this is a boot disk. The virtual machine will
+	// use the first partition of the disk for its root filesystem.
+	Boot bool `json:"boot"`
+
+	// DiskSizeGb: Specifies the size of the disk in base-2 GB.
```
Comment (ggaurav10) [inline pkg/gcp/apis/provider_spec.go:114]:
```suggestion
	// SizeGb: Specifies the size of the disk in base-2 GB.
```

File: pkg/gcp/apis/provider_spec.go [hunk 2]
```diff
@@ -22,65 +22,257 @@ const (
...
+	SizeGb int64 `json:"sizeGb"`
+
+	// DiskType: Specifies the disk type to use to create the instance. If
```
Comment (ggaurav10) [inline pkg/gcp/apis/provider_spec.go:117]:
```suggestion
	// Type: Specifies the disk type to use to create the instance. If
```

File: pkg/gcp/apis/provider_spec.go [hunk 3]
```diff
@@ -22,65 +22,257 @@ const (
...
+	// can use either NVME or SCSI. For performance characteristics of SCSI
+	// over NVMe, see Local SSD performance.
+	//
+	// Possible values:
+	//   "NVME"
+	//   "SCSI"
+	// This is only applied when the disk type is "SCRATCH" currently
+	Interface string `json:"interface"`
+
+	// SourceImage: The source image to create this disk. When creating a
```
Comment (ggaurav10) [inline pkg/gcp/apis/provider_spec.go:149]:
```suggestion
	// Image: The source image to create this disk. When creating a
```

**Comment by ggaurav10** [comment]
1 [test](https://github.com/gardener/machine-controller-manager-provider-gcp/blob/master/pkg/gcp/machine_controller_test.go#L230-L242) is [failing](https://concourse.ci.gardener.cloud/builds/392981). I think it's because one of the error message changed from MCM.

**Comment by prashanth26** [comment]
Tests seem to be passing here again later? -https://concourse.ci.gardener.cloud/teams/gardener/pipelines/machine-controller-manager-provider-gcp-master/jobs/master-pull-request-job/builds/5

**Pull request merged**